engine/src/main/resources/org/terasology/include/globalFunctionsFragIncl.glsl

```diff
@@ -147,16 +147,15 @@ float calcPcfShadowTerm(sampler2D shadowMap, float lightDepth, vec2 texCoord, fl
 float calcVolumetricFog(vec3 fogWorldPosition, float volumetricHeightDensityAtViewer, float globalDensity, float heightFalloff) {
     vec3 cameraToFogWorldPosition = -fogWorldPosition;
 
-    float fogInt = length(cameraToFogWorldPosition) * volumetricHeightDensityAtViewer;
+    float totalFogToSample = length(cameraToFogWorldPosition) * globalDensity * volumetricHeightDensityAtViewer;
+    
     const float slopeThreshold = 0.01;
-
-    if (abs(cameraToFogWorldPosition.y) > slopeThreshold)
-    {
-        float t = heightFalloff * cameraToFogWorldPosition.y;
-        fogInt *= (1.0 - exp(-t)) / t;
+    float heightDensityFactor = heightFalloff * cameraToFogWorldPosition.y;
+    if (abs(heightDensityFactor) > slopeThreshold) {
+        totalFogToSample *= (1.0 - exp(-heightDensityFactor)) / heightDensityFactor;
     }
 
-    return exp(-globalDensity * fogInt);
+    return 1 - exp(-totalFogToSample);
```

Comment:
I'm feeling this part of the code could use some tender-loving-care: the math of the individual formulas is easy enough, but the ensemble is not really easy to follow for me, starting from the slopeThreshold constant until the return line.

I mean: I can run Terasology and see that it now does something it didn't do before, but I have no idea if what I see is the intended effect. It almost seems like the intended effect here is some kind of falloff of the fog effect over altitude for example. But I'm not seeing that right now when I toggle the effect. Do I have to go much higher perhaps? I don't know.

This of course is not to say it's your fault @4Denthusiast, it's just code we inherited from previous developers and you just brought it into the spotlight. It's just something the "rendering department" will have to deal with at some point, in the context of the Shaderland Refactoring. 

For the time being I think your PR gets the job done and after some last checks it should be mergeable.

Reply:
The fog falloff over altitude does work, it's just set to a small value so it's not very noticeable, and the fog density at the viewer's position is fixed even if you move up or down. (That's what fogDensityAtViewer is for.)